### PR TITLE
Added CUB dependency to Aluminum

### PR DESCRIPTION
**Pull request by bvanessen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/aluminum/package.py [hunk 1]
```diff
@@ -60,6 +62,10 @@ def cmake_args(self):
             args.extend([
                 '-DALUMINUM_ENABLE_MPI_CUDA:BOOL=%s' % ('+ht' in spec)])
 
+        if '@:0.1,0.6.0:':
+            args.extend([
+                '-DCUB_DIR:FILEPATH=%s' % spec['cub'].prefix.include])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aluminum/package.py]:
```suggestion
            args.append(
                '-DCUB_DIR:FILEPATH=%s' % spec['cub'].prefix.include)
```

**Review by adamjstewart — APPROVED** [review]
Would you or anyone else like to be maintainers for this package?

File: var/spack/repos/builtin/packages/aluminum/package.py [hunk 2]
```diff
@@ -57,8 +61,12 @@ def cmake_args(self):
                 '-DALUMINUM_ENABLE_MPI_CUDA_RMA:BOOL=%s' %
                 ('+cuda_rma' in spec)])
         else:
+            args.extend(
+                '-DALUMINUM_ENABLE_MPI_CUDA:BOOL=%s' % ('+ht' in spec))
+
+        if '@:0.1,0.6.0:':
             args.extend([
-                '-DALUMINUM_ENABLE_MPI_CUDA:BOOL=%s' % ('+ht' in spec)])
+                '-DCUB_DIR:FILEPATH=%s' % spec['cub'].prefix.include])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aluminum/package.py]: `list.extend` takes a list, `list.append` takes a single argument. I think you're using both wrong in multiple places in this package.

File: var/spack/repos/builtin/packages/aluminum/package.py (hunk 1 above)
Comment (bvanessen) [reply var/spack/repos/builtin/packages/aluminum/package.py]: I just fixed this.

File: var/spack/repos/builtin/packages/aluminum/package.py (hunk 2 above)
Comment (bvanessen) [reply var/spack/repos/builtin/packages/aluminum/package.py]: Fixed.

**Comment by bvanessen** [comment]
> Would you or anyone else like to be maintainers for this package?

@adamjstewart I added myself as the maintainer.

**Comment by bvanessen** [comment]
@adamjstewart Can we get this merged please?

**Comment by adamjstewart** [comment]
It looks like there is a merge conflict now. Can you resolve that and ping me when the tests finish?

**Comment by bvanessen** [comment]
@adamjstewart I have rebased and fixed the merge conflicts

**Pull request merged**